js/src/client.ts

```diff
@@ -2921,4 +2940,566 @@ export class Client {
+        headers: this.headers,
+        signal: AbortSignal.timeout(this.timeout_ms),
+        ...this.fetchOptions,
+      }
+    );
+    const json = await res.json();
+    if (!res.ok) {
+      const detail =
+        typeof json.detail === "string"
+          ? json.detail
+          : JSON.stringify(json.detail);
+      const error = new Error(
+        `Error ${res.status}: ${res.statusText}\n${detail}`
+      );
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (error as any).statusCode = res.status;
+      throw error;
+    }
+    return json;
+  }
+
+  public async listPrompts(options?: {
+    limit?: number;
+    offset?: number;
+    isPublic?: boolean;
+    isArchived?: boolean;
+    sortField?: PromptSortField;
+    sortDirection?: "desc" | "asc";
+    query?: string;
+  }): Promise<ListPromptsResponse> {
+    const params: Record<string, string> = {
+      limit: (options?.limit ?? 100).toString(),
+      offset: (options?.offset ?? 0).toString(),
+      sort_field: options?.sortField ?? "updated_at",
+      sort_direction: options?.sortDirection ?? "desc",
+      is_archived: (!!options?.isArchived).toString(),
+    };
+
+    if (options?.isPublic !== undefined) {
+      params.is_public = options.isPublic.toString();
+    }
+
+    if (options?.query) {
+      params.query = options.query;
+    }
+
+    const queryString = new URLSearchParams(params).toString();
+
+    const response = await this.caller.call(
+      fetch,
+      `${this.apiUrl}/repos/?${queryString}`,
+      {
+        method: "GET",
+        headers: this.headers,
+        signal: AbortSignal.timeout(this.timeout_ms),
+        ...this.fetchOptions,
+      }
+    );
+
+    const res = await response.json();
+
+    return {
+      repos: res.repos.map((result: any) => ({
+        owner: result.owner,
+        repoHandle: result.repo_handle,
+        description: result.description,
+        id: result.id,
+        readme: result.readme,
+        tenantId: result.tenant_id,
+        tags: result.tags,
+        isPublic: result.is_public,
+        isArchived: result.is_archived,
+        createdAt: result.created_at,
+        updatedAt: result.updated_at,
+        originalRepoId: result.original_repo_id,
+        upstreamRepoId: result.upstream_repo_id,
+        fullName: result.full_name,
+        numLikes: result.num_likes,
+        numDownloads: result.num_downloads,
+        numViews: result.num_views,
+        likedByAuthUser: result.liked_by_auth_user,
+        lastCommitHash: result.last_commit_hash,
+        numCommits: result.num_commits,
+        originalRepoFullName: result.original_repo_full_name,
+        upstreamRepoFullName: result.upstream_repo_full_name,
+      })),
+      total: res.total,
+    };
+  }
+
+  public async getPrompt(promptIdentifier: string): Promise<Prompt | null> {
+    const [owner, promptName, _] = parsePromptIdentifier(promptIdentifier);
+    const response = await this.caller.call(
```

Comment:
I think we should use `raiseForStatus` after receiving every response to properly test of `!response.ok`